spec.html

```diff
@@ -20773,6 +21664,15 @@ <h1>Runtime Semantics: Evaluation</h1>
         1. Set _value_.[[SourceText]] to the source text matched by |ClassExpression|.
         1. Return _value_.
       </emu-alg>
+      <emu-grammar>ClassElementName : PrivateIdentifier</emu-grammar>
+      <emu-alg>
+        1. Let _privateIdentifier_ be StringValue of |PrivateIdentifier|.
+        1. Let _privateEnvRec_ be the running execution context's PrivateEnvironment.
+        1. Let _names_ be _privateEnvRec_.[[Names]].
+        1. Assert: _names_ contains exactly one Private Name whose [[Description]] is _privateIdentifier_.
+        1. Let _privateName_ be the Private Name in _names_ whose [[Description]] is _privateIdentifier_.
+        1. Return _privateName_.
```

Comment:
is the repetition needed here (because the List could contain elements that are not a Private Name), or could this be:
```suggestion
        1. Return the sole element of _names_.
```

Reply:
The list only contains private names, but the others are guaranteed to have a different [[Description]].

Reply:
The previous assert, then, should ensure that the sole element of `names` is the right one?

Reply:
I don't know where you are getting "sole element" from. There can be many elements in the list, but there's only one whose [[Description]] is _privateIdentifier_.

Reply:
The previous line says "Assert: _names_ contains exactly one Private Name whose [[Description]] is _privateIdentifier_.".

I read that as, the list contains one Private Name, and its [[Description]] is as expected.

It sounds like the intention is "it contains many Private Names, but only one of them has a [[Description]] as expected"?

Reply:
Ah, yes, I intended the second reading.

Reply:
Is there another way this could be phrased to avoid the confusion?

Reply:
Rephrased to `Assert: exactly one element of _names_ is a Private Name whose [[Description]] is _privateIdentifier_.`

Reply:
Thanks, that definitely makes me read it matching your intention.